src/gui/mainwindow.cpp

```diff
@@ -681,9 +681,22 @@ void MainWindow::on_actionLock_triggered()
     hide();
 }
 
-void MainWindow::handleRSSUnreadCountUpdated(int count)
+void MainWindow::handleRSSUnreadCountUpdated(int)
 {
-    m_tabs->setTabText(m_tabs->indexOf(m_rssWidget), tr("RSS (%1)").arg(count));
+    if (m_rssUnreadUpdateEnqueued)
+        return;
+
+    m_rssUnreadUpdateEnqueued = true;
+    QTimer::singleShot(100, this, [this]()
+    {
+        m_rssUnreadUpdateEnqueued = false;
+
+        if (!m_rssWidget)
```

Comment:
Why not use `m_rssWidget` as a context to avoid this check?

Reply:
m_rssWidget is a QPointer, so I thought it's not owned by `this`.

Reply:
I mean `QTimer::singleShot(100, m_rssWidget, [this]()` so connection will be destroyed when `m_rssWidget` is destroyed and you don't need check it inside lambda body.

Reply:
what if `this` is destroyed but `m_rssWidget` not

> m_rssWidget is a QPointer, so I thought it's not owned by this.

checking a `QPointer` before it's use is recommended.
I didn't look into who owns `m_rssWidget`

Reply:
MainWindow owns (should own) all child widgets "by design". QPointers are just misused in legacy code. However, since we've stumbled upon this problem and you're bumping into it, you can wait for me or someone else to fix it.

Reply:
> you can wait for me or someone else to fix it.

extra code can be removed later on when someone will fix this problem.

Reply:
> extra code can be removed later on when someone will fix this problem.

Extra code can be removed now since fixing the problem doesn't really affect it. As I said above MainWindow owns child widgets so it is destroyed after them.